### PR TITLE
Feature/add hero animations

### LINKED ISSUE #20: Add remaining feature animations
We have designs and prototypes for the remaining animations. They are as follows:

- [Hero section text](https://www.figma.com/proto/3OU7t2Pk3t1pO5eaVi3z0M/Kedro-Prototypes?node-id=20%3A331&scaling=min-zoom&page-id=20%3A140&starting-point-node-id=20%3A331&show-proto-sidebar=1)
- [Hero section buttons](https://www.figma.com/proto/3OU7t2Pk3t1pO5eaVi3z0M/Kedro-Prototypes?node-id=28%3A583&scaling=min-zoom&page-id=20%3A140&starting-point-node-id=20%3A331&show-proto-sidebar=1)
  - Click the "Get Started" and "Take The Tutorial" buttons here to see the animations
-  [Mobile menu](https://www.figma.com/proto/3OU7t2Pk3t1pO5eaVi3z0M/Kedro-Prototypes?node-id=126%3A90&scaling=min-zoom&page-id=124%3A89&starting-point-node-id=126%3A90)
   - This is already in place, though I think we can finesse it a bit

### PR DESCRIPTION
**Pull request by tynandebold**
## Description

Resolves #20.

## Development notes

Wow, what a pain. The animations are sorted now. They don't match exactly 100%, but they're very close. I still don't have that little rocket ship icon, as discussed in the standup today, but we can add that later.

I've further finessed the mobile menu animation as well as added in the favicon, which required adding a [custom document page](https://nextjs.org/docs/advanced-features/custom-document). That's a small change though.

Lastly, I added a `useWindowSize` hook to our `/utils/hooks/` folder. I need to be able to determine what width to use for the hero animation when we're in our mobile breakpoint, and this is how I've done it.

## QA notes

Please reference [the issue](https://github.com/kedro-org/kedro-website/issues/20) to view the specific prototypes for the animations to check against and have a look at the website. Hover over the hero buttons and open/close the mobile menu.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Added tests to cover my changes, where applicable

### CODE REVIEW

File: components/hero/hero.tsx [hunk 1]
```diff
@@ -1,26 +1,144 @@
...
 import style from './hero.module.scss';
 
+const headerText = [
+  { width: 465, word: 'Maintainable' },
+  { width: 298, word: 'Modular' },
+];
+const MOBILE_BREAKPOINT = 819;
+const BUTTON_HEIGHT = 48; // 3rem.
+
+function computeVariants(isMobile: boolean) {
```
Comment (studioswong) [inline components/hero/hero.tsx:17]: perhaps `computeAnimationVariants` would be a more descriptive name?

File: components/hero/hero.module.scss [hunk 2]
```diff
@@ -45,6 +49,50 @@
...
+    }
+
+    > span:last-of-type {
+      margin-left: 0.6rem;
+    }
+  }
+}
+
+.animatingWord {
+  height: 90px;
```
Comment (studioswong) [inline components/hero/hero.module.scss:81]: Let's rem some of these up :)

**Review by studioswong — COMMENTED** [review]
the animation looks rad - well done!

one observations - the length of the buttons on seems to be of variable length ( see below) as compared to a unified length in the design

Mobile view:
image

responsive view:
image

image

please see some further comments below.

Otherwise, well done!

File: components/hero/hero.module.scss (hunk 2 above)
Comment (tynandebold) [reply components/hero/hero.module.scss:81]: Ah yes. Fixed.

**Comment by tynandebold** [comment]
Thanks for the review, Susanna :) I've made all your fixes now.

**Review by rashidakanchwala — APPROVED** [review]
look's great @tynandebold 

I just noticed one minor thing, the maintainable text is cut off by a bit in the Ipad Air breakpoint

Screenshot 2022-04-01 at 13 55 21

**Comment by tynandebold** [comment]
> look's great @tynandebold
> 
> I just noticed one minor thing, the maintainable text is cut off by a bit in the Ipad Air breakpoint
> 
> Screenshot 2022-04-01 at 13 55 21

Good catch :) I've fixed it now.

**Review by studioswong — APPROVED** [review]
Nice one!

**Pull request merged**